### ISSUE TITLE
List not updating when using fastjoin

### ISSUE THREAD

**Issue by dottodot** [feathersjs-ecosystem/feathers-reactive#105]
I have the following mongoose model 

```
  const orderComment = new Schema(
    {
      comment: 'string',
      order: {
        type: ObjectId,
        ref: 'order'
      },
      line_item: {
        type: ObjectId,
        ref: 'lineItem'
      },
      owner: {
        type: ObjectId,
        ref: 'users'
      }
    },
    {
      timestamps: true
    }
  );
```
and I'm populating the `line_item` using fastjoin i.e

```
  after: {
    all: [fastJoin(resolver)],
    find: [],
    get: [],
    create: [],
    update: [],
    patch: [],
    remove: []
  },

  joins: {
    line_item: () => async (post, context) =>
      !post.line_item
        ? null
        : (post.line_item = await context._loaders.items._id.load(
            post.line_item
          ))
  }
```
How ever when I create a new comment with the following data the list doesn't automatically update
```
{
  comment: 'my comment';
  line_item: '5a7ef92cfadfee10fb85d306';
  order: '5a7ef92bfadfee10fb85d305';
}
```
if I exclude the line_item property or remove the population is does work, so I assume it's because what I'm posting is formatted different to the list, but no idea how you get around this.

**Comment by j2L4e** [comment]
What's your query?

**Comment by dottodot** [comment]
Why doesn't my list automatically update when I add a item and fastjoin is used to populate one of the values.

**Comment by j2L4e** [comment]
Let me rephrase: How do you query the data? What does your query object look like?

**Comment by dottodot** [comment]
Ah right sorry, it's

```
    const query: any = {
      order: this.orderId,
      line_item: line_item,
      $sort: { createdAt: -1 },
      $limit: 25
    };

    return this.feathers
      .service('order-comment')
      .watch()
      .find({
        query: query
      });
```

**Comment by j2L4e** [comment]
The smart strategy works like this:

a) Run the query as usual and return the result as first item on the Observable.
b) on any create event (and all the other events) determine if the created object would have matched the query from a) if it had already existed back then. If yes, insert it into the set and reemit on the Observable.

Now, as you have populated line_item, the id you queried for doesn't match the object that is returned:

```
{
  comment: 'my comment';
  line_item: {
    foo: 1,
    bar: 2
  },
  order: '5a7ef92bfadfee10fb85d305';
}
```

Here's a workaround (untested) that might do what you need:

```
const firstQuery = {
    order: this.orderId,
    line_item: line_item,
    $sort: { createdAt: -1 },
    $limit: 25
};

const secondQuery = {
    order: this.orderId,
    line_item: {
        _id: line_item
    },
    $sort: { createdAt: -1 },
    $limit: 25
};

this.feathers
    .service('order-comment')
    .watch()
    .find({
        query: secondQuery
    })
    .pipe(
      switchMapTo(
        this.feathers.service('order-comment').find({ query: firstQuery })
      )
    );
```

It will basically requery with the first query on any event mathcing the second query.

I'd suggest not populating line_item into itself (i.e. not replacing line_item), but rather populate it into a different field:

```
{
  comment: 'my comment',
  line_item: '5a7ef92cfadfee10fb85d306',
  order: '5a7ef92bfadfee10fb85d305',
}

// populated:

{
  comment: 'my comment',
  line_item: '5a7ef92cfadfee10fb85d306',
  line_item_data: {
    foo: 1,
    blah: 2
  },
  order: '5a7ef92bfadfee10fb85d305',
}
```

@daffl we should think about providing the possibility to pass in two query objects, one for the initial query and the other one for event matching.

**Comment by dottodot** [comment]
I've gone with the second solution which works, probably seems a better idea than making 2 queries.

Would be nice if there was an option to get round it built in though. Anyway thanks for your help.

**Comment by j2L4e** [comment]
Glad I could help :tada:

**Comment by j2L4e** [comment]
reopening, because the issue wasn't actually solved :roll_eyes:
